src/Sarif.Viewer.VisualStudio/ErrorList/SinkWrapper.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) Microsoft. All rights reserved. 
+// Licensed under the MIT license. See LICENSE file in the project root for full license information. 
+
+using System;
+
+using Microsoft.VisualStudio.Shell.TableManager;
+
+namespace Microsoft.Sarif.Viewer.ErrorList
+{
+    // Simple wrapper class that holds an ITableDataSink, and raises an event when
+    // Visual Studio disposes it.
+    //
+    // When a client such as the viewer calls ITableManager.AddSource to add a new
+    // table data source, Visual Studio calls ITableDataSource.Subscribe on the source,
+    // providing a "sink": an object that implements ITableDataSink. In return, VS
+    // receives a disposable object associated with the sink. When the source is
+    // removed, VS calls Dispose on this object, at which point the client must
+    // stop making calls to that sink.
+    //
+    // To accomplish this, SinkWrapper raises an event when VS disposes it. The
+    // receiver must remove the associated sink from the list of sinks it maintains.
+    // See SarifTableDataSource.TableSink_Disposed.
+    internal class SinkWrapper : IDisposable
```

Comment:
>SinkWrapper [](start = 19, length = 11)

This class replaces `SarifTableDataSink`, which did more than it needed to in wrapping the calls to `IDataTableSink`.